src/models/tools/geometry/jxg-comment.ts

```diff
@@ -0,0 +1,137 @@
+import { JXGChangeAgent, JXGProperties } from "./jxg-changes";
+import { objectChangeAgent } from "./jxg-object";
+import { values } from "lodash";
+import { isMovableLine } from "./jxg-movable-line";
+import { isPolygon } from "./jxg-polygon";
+import { isBoard } from "./jxg-board";
+import { isPoint } from "./jxg-point";
+
+export const isCommentType = (v: any) => v && v.getAttribute("clientType") === "comment";
+
+export const isComment = (v: any) => isCommentType(v) && (v instanceof JXG.Text) && (v.elType === "text");
+
+export const getAnchor = (comment: JXG.Text) => {
+  if (!isComment) return;
+  const ancestors = values(comment.ancestors);
+  return ancestors.length === 1
+    ? ancestors[0]
+    : ancestors.find(elem => isMovableLine(elem) || isPolygon(elem));
```

Comment:
The relationship between the number and type of ancestors is not clear to me and so warrants a comment.

Reply:
Wow good point I'm not sure what this function is doing - it's never actually called so I'll do you one better and delete it 😬